sklearn/cluster/_agglomerative.py

```diff
@@ -743,6 +743,11 @@ class AgglomerativeClustering(ClusterMixin, BaseEstimator):
 
         .. versionadded:: 0.21
 
+    return_distance : bool, default=False
```

Comment:
I would prefer `compute_distances` since the distance will be stored as an attribute.